src/zoom_func.h

```diff
@@ -68,6 +68,26 @@ static inline int UnScaleGUI(int value)
 	return UnScaleByZoom(value, ZOOM_LVL_GUI);
 }
 
+/**
+ * Scale zoom level relative to GUI zoom.
+ * @param value zoom level to scale
+ * @return scaled zoom level
+ */
+static inline ZoomLevel ScaleZoomGUI(ZoomLevel value)
+{
+	return ZoomLevel(value + (ZOOM_LVL_GUI - ZOOM_LVL_OUT_4X));
+}
+
+/**
+ * UnScale zoom level relative to GUI zoom.
+ * @param value zoom level to scale
+ * @return un-scaled zoom level
+ */
+static inline ZoomLevel UnScaleZoomGUI(ZoomLevel value)
+{
+	return ZoomLevel(value - (ZOOM_LVL_GUI - ZOOM_LVL_OUT_4X));
```

Comment:
Hmm, wouldn't these functions need some clamping to the range [ZOOM_LVL_BEGIN, ZOOM_LVL_END)?
I am also unsure what happens, if the max-zoom-in level is limited in the settings, maybe that is a limiting factor as well?

Reply:
I've updated the code to clamp to the range `[ZOOM_LVL_MIN, ZOOM_LVL_MAX]`. Applying the settings is not something that I feel these methods should do: that's up to call-site to apply the settings (if needed).